api/v1beta1/keystoneapi_types.go

```diff
@@ -180,6 +180,15 @@ type KeystoneAPISpecCore struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	// TLS - Parameters related to the TLS
 	TLS tls.API `json:"tls,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=false
+	// Enablement of Federation configuration
+	EnableFederation bool `json:"enableFederation"`
+
+	// +kubebuilder:validation:Optional
+	// +OIDCFederation - parameters to configure keystone for OIDC federation
+	OIDCFederation KeystoneFederationSpec `json:"oidcFederation,omitempty"`
```

Comment:
might want to do this a ptr so that it won't show up with all the empty values?

```
apiVersion: keystone.openstack.org/v1beta1
kind: KeystoneAPI
metadata:
  creationTimestamp: "2024-11-18T19:38:57Z"
  finalizers:
  - openstack.org/keystoneapi
  - openstack.org/keystoneservice-nova
  - openstack.org/keystoneservice-glance
  - openstack.org/keystoneservice-ceilometer
  - openstack.org/keystoneservice-swift
  - openstack.org/keystoneendpoint-glance-default-external
  - openstack.org/keystoneendpoint-glance-default-internal
  - openstack.org/keystoneendpoint-swift
  - openstack.org/keystoneservice-barbican
  - openstack.org/keystoneservice-cinderv3
  - openstack.org/keystoneendpoint-barbican-api
  - openstack.org/keystoneendpoint-cinderv3
  - openstack.org/keystoneservice-neutron
  - openstack.org/keystoneservice-placement
  - openstack.org/keystoneendpoint-neutron
  - openstack.org/keystoneendpoint-placement
  - openstack.org/keystoneendpoint-nova
  generation: 2
  name: keystone
  namespace: openstack
  ownerReferences:
  - apiVersion: core.openstack.org/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: OpenStackControlPlane
    name: openstack-galera
    uid: 54409ae3-a561-43b7-9251-603437046eb3
  resourceVersion: "70386"
  uid: fbc4eab5-72b6-485e-b9b9-6536919d4c4b
spec:
  adminProject: admin
  adminUser: admin
  containerImage: quay.io/podified-antelope-centos9/openstack-keystone@sha256:198eb79012e1b824b4092e4907f0565051f8514a7ca0d4606e9ce7a3d8439895
  databaseAccount: keystone
  databaseInstance: openstack
  enableFederation: false
  enableSecureRBAC: true
  fernetMaxActiveKeys: 5
  fernetRotationDays: 1
  memcachedInstance: memcached
  oidcFederation:
    keystoneFederationIdentityProviderName: ""
    oidcCacheType: ""
    oidcClaimDelimiter: ""
    oidcClaimPrefix: ""
    oidcClientID: ""
    oidcIntrospectionEndpoint: ""
    oidcMemCacheServers: ""
    oidcPassClaimsAs: ""
    oidcPassUserInfoAs: ""
    oidcProviderMetadataURL: ""
    oidcResponseType: ""
    oidcScope: ""
    remoteIDAttribute: ""
```

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_keystone-operator/479/pull-ci-openstack-k8s-operators-keystone-operator-main-keystone-operator-build-deploy-kuttl/1858575042430898176/artifacts/keystone-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-3d1f620ae3617fbd6371f55aa948b34e9daf31a461d3c557cf1a7c2db38d0385/namespaces/openstack/crs/keystoneapis.keystone.openstack.org/keystone.yaml

Reply:
just change it to 
```
OIDCFederation *KeystoneFederationSpec `json:"oidcFederation,omitempty"`
```
probably needs some check for it bein nil where you right now just use it.

Reply:
You can see how I'm doing it with the PKCS11 settings in barbican in https://github.com/openstack-k8s-operators/barbican-operator/pull/168

Reply:
Done